hdr.go

```diff
@@ -372,7 +388,7 @@ func (h *Histogram) ValueAtPercentiles(percentiles []float64) (values map[float6
 	total := int64(0)
 	currentQuantileSlicePos := 0
 	i := h.iterator()
-	for currentQuantileSlicePos < totalQuantilesToCalculate && i.nextCountAtIdx() {
+	for currentQuantileSlicePos < totalQuantilesToCalculate && i.nextCountAtIdx(h.totalCount) {
```

Comment:
this looks tricky, can you add a comment explaining how h.totalCount can be different from i.h.totalCount?

Reply:
@ahothan this was added to avoid checking outside/inside twice the same condition:
> condition on hdr.go#L626 taking ~11% of cpu-time: if i.countToIdx >= i.h.totalCount . Notice that we're doing a more restrictive check at hdr.go#L340 if total >= countAtPercentile {, meaning we can completely avoid this condition check.

> how h.totalCount can be different from i.h.totalCount?
it can't but in this manner we can have a more restrictive check and avoid 2X the price of checking inside `f i.countToIdx >= i.h.totalCount` and outside `if total >= <value between 0 and totalCount> `